ros2_generate_interface_docs/ros2_generate_interface_docs/main.py

```diff
@@ -129,14 +134,19 @@ def main(argv=sys.argv[1:]):
     services = get_service_interfaces()
     actions = get_action_interfaces()
 
+    if len(args.packages_select) > 0:
+        messages = {k: v for k, v in messages.items() if k in args.packages_select}
+        services = {k: v for k, v in services.items() if k in args.packages_select}
+        actions = {k: v for k, v in actions.items() if k in args.packages_select}
```

Comment:
Why filter the results after all interfaces have been collected? The above function offer an argument to narrow down the packages already.